### PR TITLE
Another BLTouch retry place (replaces PR #6108)

### PR DESCRIPTION
**Pull request by Bob-the-Kuhn**
This PR replaces PR #6108. - Couldn't get a truthful compare on it after resolving a conflict & doing a rebase.

----

While testing UBL I discovered that the set_bltouch_deployed routine also needed the reset-deploy-stow sequence if there is an error.

I also added a 1.5 second delay after the stow because sometimes the reset will take the BLTouch signal back to normal but 3/4 of a second later it would go back into the error state.

I also added a delay just before calling stop() to allow the error message to get out the serial port before the machine halts.

----

This is from PR #6086 

I had a sticky BLTouch that wasn't clearing with the current code. I tried more deploy/stow cycles and various delays with no luck. Looks like if it's going to clear then a single reset-deploy-stow cycle is sufficient. If it's still in error after one cycle then there's a real problem.

I did notice that the BLTouch would go back into it's error state about 750mS after the S160 cleared it so I'll add a delay. Unfortunately the 750mS was just what I saw the most of. Some were more than 2 seconds. I think I'll just put in a 1.5 second delay which should catch most of the real problems.

### CODE REVIEW

File: Marlin/Marlin_main.cpp [hunk 1]
```diff
@@ -2062,6 +2062,21 @@ static void clean_up_after_endstop_or_probe_move() {
     }
 
     void set_bltouch_deployed(const bool deploy) {
+      if (deploy && TEST_BLTOUCH()) {      // If BL-Touch says it's triggered
+        bltouch_command(BLTOUCH_RESET);    // try to reset it.
+        bltouch_command(BLTOUCH_DEPLOY);   // Also needs to deploy and stow to
+        bltouch_command(BLTOUCH_STOW);     // clear the triggered condition.
+        delay(1500);                       // wait for internal self test to complete
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp]: We have to use `safe_delay` to keep everything in Marlin running. 3 more instances below.

File: Marlin/Marlin_main.cpp [hunk 2]
```diff
@@ -2092,13 +2107,17 @@ static void clean_up_after_endstop_or_probe_move() {
...
         bltouch_command(BLTOUCH_RESET);    // try to reset it.
-        set_bltouch_deployed(true);        // Also needs to deploy and stow to
-        set_bltouch_deployed(false);       // clear the triggered condition.
+        bltouch_command(BLTOUCH_DEPLOY);   // Also needs to deploy and stow to
+        bltouch_command(BLTOUCH_STOW);     // clear the triggered condition.
+        delay(1500);                       // wait for internal self test to complete
+                                           //   measured completion time was 0.65 seconds
+                                           //   after reset, deploy & stow sequence
         if (TEST_BLTOUCH()) {              // If it still claims to be triggered...
+          SERIAL_EOL;
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp]: Why the extra `SERIAL_EOL`? Perhaps we should just do this, if there's concern about error-detection:
```cpp
const char errormagic[] PROGMEM = "\nError:";
```

File: Marlin/Marlin_main.cpp [hunk 3]
```diff
@@ -2092,13 +2107,17 @@ static void clean_up_after_endstop_or_probe_move() {
...
           SERIAL_ERROR_START;
           SERIAL_ERRORLNPGM(MSG_STOP_BLTOUCH);
+          delay(350);   // wait until message is out the serial port before crashing
           stop();                          // punt!
-          return true;
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp]: The `stop()` doesn't "crash," but just halts the machine and waits for `M999`. So `return true` is needed here to return the fail condition and exit the function.

File: Marlin/Marlin_main.cpp [hunk 4]
```diff
@@ -2062,6 +2062,21 @@ static void clean_up_after_endstop_or_probe_move() {
...
+        bltouch_command(BLTOUCH_STOW);     // clear the triggered condition.
+        delay(1500);                       // wait for internal self test to complete
+                                           //   measured completion time was 0.65 seconds
+                                           //   after reset, deploy & stow sequence
+        if (TEST_BLTOUCH()) {              // If it still claims to be triggered...
+          SERIAL_EOL;
+          SERIAL_ERROR_START;
+          SERIAL_ERRORLNPGM(MSG_STOP_BLTOUCH);
+          delay(350);   // wait until message is out the serial port before crashing
+          stop();                          // punt!
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp]: After `stop()` do `return true` to exit with fail.

**Comment by Bob-the-Kuhn** [comment]
Removed the delays - they're being done under PR #6190.

Also moved the return statement to the correct place.

**Comment by Bob-the-Kuhn** [comment]
All review comments have been incorporated.

No planned changes.

**Comment by Roxy-3D** [comment]
@Bob-the-Kuhn   One more item to discuss:  https://github.com/MarlinFirmware/Marlin/issues/6262

**Pull request merged**